File: test-plugins/org.yakindu.sct.generator.cpp.test/gtests/OperationsWithoutBracesTest/OperationsWithoutBracesTestCustom.cc
/**
* Copyright (c) 2015 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include <string>
#include "gtest/gtest.h"
#include "OperationsWithoutBraces.h"
#include "OperationsWithoutBraces_OCB.h"

TEST(StatemachineTest, OperationsWithoutBracesTest) {
	OperationsWithoutBraces* statechart = new OperationsWithoutBraces();
	statechart->init();
	statechart->enter();
	
	statechart->setDefaultSCI_OCB(new DefaultCallBack());
	
	EXPECT_TRUE(statechart->isStateActive(OperationsWithoutBraces::main_region_A));
	EXPECT_TRUE(statechart->isStateActive(OperationsWithoutBraces::another_region_A));
	statechart->runCycle();
	EXPECT_TRUE(statechart->isStateActive(OperationsWithoutBraces::main_region_B));
	EXPECT_TRUE(statechart->isStateActive(OperationsWithoutBraces::another_region_B));
	statechart->runCycle();
	EXPECT_TRUE(statechart->isStateActive(OperationsWithoutBraces::main_region_C));
	EXPECT_TRUE(statechart->isStateActive(OperationsWithoutBraces::another_region_C));
	statechart->runCycle();
	EXPECT_TRUE(statechart->isStateActive(OperationsWithoutBraces::main_region_D));
	EXPECT_TRUE(statechart->isStateActive(OperationsWithoutBraces::another_region_D));

	delete statechart;
}

sc_boolean DefaultCallBack::myBoolOp(){
	return false;
}

sc_integer DefaultCallBack::myIntOp(){
	return 0;
}

sc_real DefaultCallBack::myRealOp(){
	return 0.0;
}

File: test-plugins/org.yakindu.sct.generator.cpp.test/src/org/yakindu/sct/generator/cpp/test/OperationsWithoutBracesTestCustom.java
/**
 * Copyright (c) 2014 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.generator.cpp.test;

import org.junit.Before;
import org.junit.runner.RunWith;
import org.yakindu.sct.generator.c.gtest.GTest;
import org.yakindu.sct.generator.c.gtest.GTestHelper;
import org.yakindu.sct.generator.c.gtest.GTestHelper.Compiler;
import org.yakindu.sct.generator.c.gtest.GTestRunner;

@GTest(
		sourceFile = "gtests/OperationsWithoutBracesTest/OperationsWithoutBracesTestCustom.cc",
		program = "gtests/OperationsWithoutBracesTest/OperationsWithoutBraces",
		model = "testmodels/SCTUnit/OperationsWithoutBraces.sct",
		statechartBundle = "org.yakindu.sct.test.models",
		additionalFilesToCopy = {
			"gtests/OperationsWithoutBracesTest/OperationsWithoutBraces_OCB.h"	
		},
		additionalFilesToCompile = {
			"OperationsWithoutBraces.cpp"
		}
		)
@RunWith(GTestRunner.class)
public class OperationsWithoutBracesTestCustom {

	protected final GTestHelper helper = new GTestHelper(this, Compiler.GPLUSPLUS) {
	};

	@Before
	public void setUp() {
		helper.generate();
		helper.compile();
	}
}
